operations/operations.py

```diff
@@ -0,0 +1,109 @@
+from functools import partial
+
+import numpy as np
+import scipp as sc
+
+
+def mask_from_adj_pixels(mask, bank_width):
+    """
+    Checks if the adjacent pixels (in 8 directions) are masked to remove
+    any noisy pixels which are erroneously masked or unmasked compared to
+    it's neighbours
+
+    If all adj. pixels are then the pixel considered is set to True
+    If no adj. pixels are then the pixel considered is set to False
+    If surrounding pixels have a mix of True/False the val is left as-is
+
+    This function handles border pixels as if they aren't there. So that
+    the following happens:
+    ------------------------
+    |F|T|     ->      |T|T|
+    |T|T|             |T|T|
+    -----------------------
+
+    Parameters
+    ----------
+    mask: Existing mask with some positions masked
+    bank_width: The width of each bank to reshape into 2D
+
+    Returns
+    -------
+    mask: Mask copy after completing the op. described above
+
+    """
+
+    def shift(var, dim, forward, out_of_bounds):
+        fill = var[dim, 0:1].copy()
+        fill.values = np.full_like(fill.values, out_of_bounds)
+        if forward:
+            return sc.concatenate(fill, var[dim, :-1], dim)
+        else:
+            return sc.concatenate(var[dim, 1:], fill, dim)
+
+    mask = mask.copy()
+    mask = sc.reshape(mask, dims=["y", "x"], shape=(bank_width, bank_width))
```

Comment:
To make this function reusable (and not break as soon as you want to natively use X and Y throughout the workflow): I suggest to remove the `bank_width` parameter and deal with the reshaping externally. If you don't want it in the notebook, add a wrapping function?